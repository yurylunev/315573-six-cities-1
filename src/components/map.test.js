import React from "react";
import renderer from "react-test-renderer";
import CitiesMap from "./map.jsx";

jest.mock(`leaflet`, () => ({
  map: () => ({
    setView: () => jest.fn()
  }),
  tileLayer: () => ({
    addTo: () => jest.fn()
  }),
  icon: () => jest.fn(),
  marker: () => ({
    addTo: () => jest.fn()
  })
}));

const testOffers = [
  {
    mark: `offerMark1`,
    imageURL: `imageURL`,
    price: 1000,
    currency: `€`,
    rate: 100,
    name: `Name of offer 1`,
    type: `Apartment type 1`,
    gps: [52.369553943508, 4.85309666406198]
  },
  {
    mark: `offerMark2`,
    imageURL: `imageURL`,
    price: 2000,
    currency: `€`,
    rate: 200,
    name: `Name of offer 2`,
    type: `Apartment type 2`,
    gps: [52.369553943508, 4.85309666406198]
  }
];

it(`App correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<CitiesMap offers={testOffers} currentView={[52.369553943508, 4.85309666406198]} clickHandler={jest.fn()}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

