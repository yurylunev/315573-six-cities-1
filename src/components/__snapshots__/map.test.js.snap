// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<div
  id="map"
/>
`;
